### PR TITLE
Update for new DNS library

### PR DESCRIPTION
**Pull request by voynovia**
Update to work

### CODE REVIEW

File: Package.resolved [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (Bouke) [inline Package.resolved]: I'm not sure if it's good practice to include package resolutions with a library. We should be fetching the latest version of dependencies, to ensure that future versions don't break. This file prevents us from doing that.

File: Package.swift [hunk 2]
```diff
@@ -1,12 +1,21 @@
...
-        .Package(url: "https://github.com/Bouke/DNS.git", majorVersion: 0, minor: 3),
-        .Package(url: "https://github.com/IBM-Swift/BlueSocket.git", majorVersion: 0, minor: 12)
-    ]
+  name: "NetService",
+  products: [
+    .library(name: "Cifaddrs", targets: ["Cifaddrs"]),
+    .library(name: "NetService", targets: ["NetService"]),
+  ],
+  dependencies: [
+    .package(url: "https://github.com/Bouke/DNS.git", .branch("master")),
```
Comment (Bouke) [inline Package.swift]: I've published a new version (1.1.0), so we don't need to target the master branch. Please specify that version `from: "1.1.0"` instead.

File: Sources/NetService/Utils.swift [hunk 3]
```diff
@@ -183,5 +183,5 @@ func gethostname() throws -> String {
 func getLocalAddresses() -> [Socket.Address] {
     return getifaddrs()
         .filter { Int($0.pointee.ifa_flags) & Int(IFF_LOOPBACK) == 0 }
-        .flatMap { Socket.Address($0.pointee.ifa_addr) }
+      .compactMap { Socket.Address($0.pointee.ifa_addr) }
```
Comment (Bouke) [inline Sources/NetService/Utils.swift]: Indentation is off here.

File: Sources/NetService/NetServiceBrowser.swift [hunk 4]
```diff
@@ -100,21 +100,21 @@ public class NetServiceBrowser: Listener {
         }
 
         let newPointers = message.answers
-            .flatMap { $0 as? PointerRecord }
+          .compactMap { $0 as? PointerRecord }
```
Comment (Bouke) [inline Sources/NetService/NetServiceBrowser.swift]: Indentation is off here and in a few other places as well.

File: .travis.yml [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
 script: swift test
 matrix:
   include:
     - os: osx
-      osx_image: xcode8.3
+      osx_image: xcode9.4
     - os: linux
       dist: trusty
-      before_install:
-        - sudo hostname travis-ci.local
```
Comment (Bouke) [inline .travis.yml]: Why remove this? The hostname should end with `.local`, see also the travis build failing because of this.

Comment (voynovia) [reply .travis.yml]: I should not have been deleted ) Is deleted when you commit "Build against 4.1 on Travis"

File: Package.resolved (hunk 1 above)
Comment (voynovia) [reply Package.resolved]: Added to .gitignore

**Comment by Bouke** [comment]
Sorry it took a while, but thanks nevertheless!

**Pull request merged**